pulsar-transformations/src/main/java/com/datastax/oss/pulsar/functions/transforms/jstl/CustomTypeConverter.java

```diff
@@ -17,17 +17,29 @@
 
 import de.odysseus.el.misc.TypeConverter;
 import de.odysseus.el.misc.TypeConverterImpl;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
 import javax.el.ELException;
 
 /**
- * Overrides the default TypeConverter coerce for null values. This fits better with Avro's nullable
- * schemas.
+ * Overrides the default TypeConverter coerce to support null values & non-EL coercions (e.g.
+ * date/time types) schemas.
  */
-public class NullableTypeConverter implements TypeConverter {
+public class CustomTypeConverter implements TypeConverter {
   private static TypeConverterImpl typeConverter = new TypeConverterImpl();
 
   @Override
   public <T> T convert(Object o, Class<T> aClass) throws ELException {
-    return o == null ? null : typeConverter.convert(o, aClass);
+    if (o == null) {
+      return null;
+    } else if ("java.time.LocalDate".equals(aClass.getName())) {
```

Comment:
why aren't we using "==" =
`java.time.LocalDate.class == aClass`

it is far more efficient